### PR TITLE
Render all block patterns of the current page and remove async list

### LINKED ISSUE #65920: Inserter: patterns refresh when selecting different blocks
Using 6.7 beta 1, I noticed that if I have the patterns tab open for the Inserter, it seems to be regularly re-rendering the list of patterns whenever I am clicking around the page. This sometimes leads to a "no results" similar to what's described here https://github.com/WordPress/gutenberg/issues/65833

https://github.com/user-attachments/assets/6ebb5004-4098-4808-afe6-4ff769567180

I am noting this as an enhancement as something to smooth out but it feels like a bug in many ways :)

### PR DESCRIPTION
**Pull request by madhusudhand**
## What?

This is an alternate to #66053. (created a separate one because it has got a UI change)

fixes: #65920

## Why?

I don't think an async pattern list is necessary anymore, because are patterns are already paginated.
Having async list causes patterns to render multiple times, which is negatively effecting the UI performance and experience.

This PR removes the async pattern list and renders all patterns of the current page (max 20). 
Also moves the pagination block to a level up, so that It is always visible.

Why move pagination?
If not fixed at the bottom, pagination visible first at the top of the page, and pushed down once patterns render.
Not sure if that should be considered as enhancement. Hence creating a separate PR instead of updating #66053

image

## Testing Instructions

1. In the editor, open pattern inserter and observe the render performance of different categories.
2. In data views, open patterns and it should render list of patterns without any performance issues.

### CODE REVIEW

**Comment by getdave** [comment]
> Also moves the pagination block to a level up, so that It is always visible.

Does this fix a bug introduced during the 6.7 cycle?

File: packages/block-editor/src/components/inserter/block-patterns-tab/pattern-category-previews.js [hunk 1]
```diff
@@ -193,8 +193,15 @@ export function PatternCategoryPreviews( {
 						isDraggable
 						showTitlesAsTooltip={ showTitlesAsTooltip }
 						patternFilter={ patternSourceFilter }
-						pagingProps={ pagingProps }
 					/>
+					{ pagingProps && (
```
Comment (getdave) [inline packages/block-editor/src/components/inserter/block-patterns-tab/pattern-category-previews.js:197]: Unless critical to this PR, then I'd suggest we take the changes that move Pagination and put in a separate PR.

**Comment by getdave** [comment]
> max 20

I'm slightly concerned about this as I know we've suffered poor performance rendering patterns. 

I wonder if we need to look again at how asyncList is working. From my tests I see to get inconsistent results with array lengths of:

- 9
- 5
- 9

I feel there could be an underlying bug here.

**Comment by madhusudhand** [comment]
This is no longer needed, as the rendering improved in #66159
